junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/JUnit5TestableFactory.java

```diff
@@ -155,7 +155,11 @@ private Method findMethod(Class<?> clazz, String methodName, Class<?>[] paramete
 	}
 
 	private Class<?> loadClassByName(String className) {
-		return ReflectionUtils.loadClass(className).orElse(null);
+		return ReflectionUtils.loadClass(className).orElseThrow(() -> createCannotLoadClassException(className));
+	}
```

Comment:
Based on your explanation, this sounds like an actual bug.

Would you be willing to raise a report against it a separate issue or pull request?

If so, that would allow us to merge your fix into `master` rather quickly.